ScanToShop/Managers/DatabaseManager.m

```diff
@@ -7,16 +7,50 @@
 //
 
 #import "DatabaseManager.h"
+#import "AlertManager.h"
 
 @implementation DatabaseManager
 
-+ (PFFileObject *)getPFFileFromImage: (UIImage * _Nullable)image {
- 
-    if (!image) {
-        return nil;
-    }
++(BOOL) registerUserInParse:(UIViewController *)vc User:(User *)user {
+    PFUser *newUser = [PFUser new];
+    newUser.username = user.username;
+    newUser.password = user.password;
+    newUser.email = user.email;
+    newUser[@"first_name"] = user.firstName;
+    newUser[@"last_name"] = user.lastName;
+    newUser[@"image"] = [DatabaseManager getPFFileFromImageData:user.profileImageData];
     
-    NSData *imageData = UIImagePNGRepresentation(image);
+    __block BOOL success;
+    [newUser signUpInBackgroundWithBlock:^(BOOL succeeded, NSError * error) {
+        if (error) {
+            NSLog(@"Error: %@", error.localizedDescription);
+            [AlertManager loginAlert:ParseBackendError ErrorString:error.localizedDescription ViewController:vc];
+        }
+        else {
+            NSLog(@"User registered successfully");
+            [vc performSegueWithIdentifier:@"registerSegue" sender:nil];
+        }
+        success = succeeded;
+    }];
+    return success;
+}
+
++(BOOL) loginWithParse:(UIViewController *)vc Username:(NSString *)username Password:(NSString *)password {
+    __block BOOL success = NO;
```

Comment:
Same here. You need to be able to call this async. It might be working just because you have a fast connection? Unclear. It is definitely wrong though.